### PR TITLE
[DO NOT MERGE] WIP Process registration request

### PR DESCRIPTION
**Pull request by Daenyth**
Servatrice can accept Command_Register proto command and perform insert.

### CODE REVIEW

File: servatrice/src/servatrice_database_interface.cpp [hunk 1]
```diff
@@ -111,11 +111,52 @@ bool Servatrice_DatabaseInterface::usernameIsValid(const QString &user)
...
+        return false;
+
+    QSqlQuery *query = prepareQuery("insert into {prefix}_users "
+            "(name, realname, gender, password_sha512, email, country, registrationDate, active) "
+            "values "
+            "(:userName, :realName, :gender, :password_sha512, :email, :country, UTC_TIMESTAMP(), :active)");
+    query->bindValue(":userName", userName);
+    query->bindValue(":realName", realName);
+    query->bindValue(":gender", getGenderChar(gender));
+    query->bindValue(":password_sha512", passwordSha512);
```
Comment (Daenyth) [inline servatrice/src/servatrice_database_interface.cpp]: I should probably have some validation that this is correct, but if they send junk it means that probably they just won't be able to connect

Comment (ctrlaltca) [inline servatrice/src/servatrice_database_interface.cpp]: the only way to be sure the password has been hashed correctly is to hash it server-side, but that would imply the client has to send the password in clear text, that is bad.

Comment (woogerboy21) [inline servatrice/src/servatrice_database_interface.cpp]: Wasn't it said that the password is sent clear text already during authentication and given the client already sends the password clear text to authenticate users would it really be that bad to have it send the password to get hashed at registration?  

Sure its not good practice but what other options would anyone suggest?

**Comment by ZeldaZach** [comment]
@Daenyth I hope you get more time so we can work on this further during the summer

**Comment by ctrlaltca** [comment]
Closing in favor of #1101

**Pull request closed without merging**